src/Views/SimulatorLog/SimulatorLog.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FunctionComponent, useState, useEffect } from "react";
+import ReactTable from "react-table";
+import "react-table/react-table.css";
+import "./SimulatorLog.css";
+import { useSelector } from "react-redux";
+import { getMotorPower } from "../../RobotSimulator/robotSimulatorSlice";
+
+/**
+ * Component for logging the commands of the simulator.
+ */
+export const SimulatorLog: FunctionComponent = () => {
+  const [tableData, setTableData] = useState<row[]>([]);
+  const motorPower0 = useSelector(getMotorPower(0));
+  const motorPower1 = useSelector(getMotorPower(1));
```

Comment:
you are right that this isn't really scalable. Instead of building the table in this component, we can add an event log to the robotSimulatorSlice. Then retrieve the list of events with a selector.

It could look something like this:
```
interface IRobotSimulatorState {
    motors: {
      [channel: number]: number;
    };
    eventLog: Array<{eventType: "setPower", /* whatever other values you want ,*/ eventSource: "blockId" /* undefined if set via UI? */}>;
  }
  
  /**
   * Reducers for handling the state of the simulator such as whether it is executing or not.
   */
  export const robotSimulatorSlice = createSlice({
    initialState: {
      motors: {},
      eventLog: []
    } as IRobotSimulatorState,
    name: "simulator",
    reducers: {
      setPower(state, action: PayloadAction<{ channel: number; power: number, source?: string }>) {
        state.motors[action.payload.channel] = action.payload.power;
        state.eventLog.push({eventType: "setPower", eventSource: action.payload.source})
  
        return state;
      },
    },
  });
```